lib/logstash/outputs/elasticsearch/common.rb

```diff
@@ -33,9 +34,33 @@ def register
 
     # Receive an array of events and immediately attempt to index them (no buffering)
     def multi_receive(events)
+      until @template_installed.true?
+        sleep 1
+      end
       retrying_submit(events.map {|e| event_action_tuple(e)})
     end
 
+    def after_successful_connection
+      @template_installer = Thread.new do
```

Comment:
I'm a bit concerned here that a pipeline could mistakenly call register twice, causing after_successful_connection to also be called twice, therefore leaking threads as `@template_installer` will be rewritten.

Reply:
Will fix